src/core/create_compilers/RollupCompiler.ts

```diff
@@ -69,13 +88,27 @@ export default class RollupCompiler {
 	}
 
 	async get_config(mod: any) {
+
+		const onwarn = mod.onwarn || ((warning: any, handler: (warning: any) => void) => {
+			handler(warning);
+		});
+
+		mod.onwarn = (warning: any) => {
+			onwarn(warning, (warn: any) => {
+				this.warnings.push(warn);
+			});
+		};
+
 		let entry_point: string | undefined;
 
 		const that = this;
-		const sourcemap = mod.output.sourcemap;
 
-		// TODO this is hacky, and doesn't need to apply to all three compilers
-		(mod.plugins || (mod.plugins = [])).push({
+		// TODO this is hacky. refactor out into an external rollup plugin
+		(mod.plugins || (mod.plugins = [])).push(css_chunks({ injectImports: true }));
+		if (!JSON.stringify(mod.input).includes('/client.')) {
```

Comment:
wouldn't this mean that any chunks that were dynamically imported *not* from the manifest wouldn't have the `import(...)` rewriting logic applied? e.g. if a page lazily loads some component upon interaction

Reply:
This line is pretty hacky and could probably be done in a cleaner way, but basically I'm saying to apply it only to the client bundle. I was thinking that you can't apply the rewriting in the server bundle because `document` won't exist there, right?